rpcclient/local.go

```diff
@@ -0,0 +1,216 @@
+package rpcclient
+
+import (
+	"context"
+	"fmt"
+	tmbytes "github.com/lazyledger/lazyledger-core/libs/bytes"
+	tmpubsub "github.com/lazyledger/lazyledger-core/libs/pubsub"
+	tmquery "github.com/lazyledger/lazyledger-core/libs/pubsub/query"
+	rpcclient "github.com/lazyledger/lazyledger-core/rpc/client"
+	"github.com/lazyledger/lazyledger-core/rpc/core"
+	ctypes "github.com/lazyledger/lazyledger-core/rpc/core/types"
+	rpctypes "github.com/lazyledger/lazyledger-core/rpc/jsonrpc/types"
+	"github.com/lazyledger/lazyledger-core/types"
+	"github.com/lazyledger/optimint/node"
+	"time"
+)
+
+var _ rpcclient.Client = &Local{}
+
+type Local struct {
+	*types.EventBus
+	ctx      *rpctypes.Context
+}
+
+func NewLocal(node *node.Node) *Local {
+	if err := node.ConfigureRPC(); err != nil {
+		node.Logger.Error("Error configuring RPC", "err", err)
+	}
+	return &Local{
+		EventBus: node.EventBus(),
+		ctx:      &rpctypes.Context{},
+	}
+}
+
+func (l Local) ABCIInfo(ctx context.Context) (*ctypes.ResultABCIInfo, error) {
+	return core.ABCIInfo(l.ctx)
```

Comment:
I did't try this myself and maybe I'm missing some context but wouldn't this code always panic? As far as I understand most of the `core` functions reply on a global var called `env`, which is never initialized here (and hence it will be `nil`). See for example the implementation of `ABCIInfo`:


```go
// ABCIInfo gets some info about the application.
// More: https://docs.tendermint.com/master/rpc/#/ABCI/abci_info
func ABCIInfo(ctx *rpctypes.Context) (*ctypes.ResultABCIInfo, error) {
	resInfo, err := env.ProxyAppQuery.InfoSync(ctx.Context(), proxy.RequestInfo)
	if err != nil {
		return nil, err
	}
	return &ctypes.ResultABCIInfo{Response: *resInfo}, nil
}
```

The same hold for all the other methods too.

Oh, I see, it would panic anyway as the `ConfigureRPC` method does panic itself - and it would be the place where the env would be created. As you have the `proxyApp` at hand already would it make sense to set the fields that you can via `n.proxyApp.Query()` and `n.proxyApp.Mempool()`? At least they seem to be independent of the rest of the `Environment`.

Reply:
Thanks for pointing this out. Actually, I decided to stop using `rpc/core` entirely. I hope it will result in cleaner code.

For now:
 - implemented methods that depends only on `proxyApp` and `EventBus`
 - other methods throws panic
 - tried to note what's necessary for each method to be implemented